### PR TITLE
Better error message if API is not available

### PR DESCRIPTION
**Pull request by Ulydev**
Screenshot 2022-09-13 at 15 33 45
Screenshot 2022-09-13 at 15 32 44

### CODE REVIEW

File: src/components/ErrorModal.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const ErrorModal: FunctionComponent<{
+    error: string
+    isOpen: boolean
+    onRequestClose: () => void
+}> = ({ error, isOpen, onRequestClose }) => (
+    <Modal
+        isOpen={isOpen}
+        onRequestClose={onRequestClose}
+        contentLabel="Example Modal"
```
Comment (brandonleung) [inline src/components/ErrorModal.tsx:16]: placeholder?

File: src/components/ErrorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        overlayClassName={classnames(
```
Comment (brandonleung) [inline src/components/ErrorModal.tsx:17]: no need to use `classnames` here unless there's something variable or we're overwriting the styling

File: src/components/setchecker/ParaSetChecker.tsx [hunk 3]
```diff
@@ -109,33 +131,45 @@ const ParaSetChecker: FunctionComponent<{}> = () => {
...
-                provider
-            );
-            setCachedParasets(Object.assign({}, ...cachedParasets.map(x => ({[x.setName]: x.amount}))));
-
-            const cachedCardsBalance = await getCachedCardsBalance(cachedParasets);
-            setCachedParasetCardTotal(Object.values(cachedCardsBalance).reduce((partialSum, a) => partialSum + a, 0));
+
+            const cachedParasets = await getCachedParasets(account, provider)
+            setCachedParasets(
+                Object.assign(
```
Comment (brandonleung) [inline src/components/setchecker/ParaSetChecker.tsx:140]: regular spread operator is more modern compared to `Object.assign`, but this is just a nit

File: src/components/setchecker/ParaSetChecker.tsx [hunk 4]
```diff
@@ -109,33 +131,45 @@ const ParaSetChecker: FunctionComponent<{}> = () => {
...
+            setCachedParasetCardTotal(
+                Object.values(cachedCardsBalance).reduce(
+                    (partialSum, a) => partialSum + a,
+                    0
+                )
+            )
         }
 
-        getCachedParasetBalances(account);
+        getCachedParasetBalances(account)
```
Comment (brandonleung) [inline src/components/setchecker/ParaSetChecker.tsx:157]: haha reverse lint?

**Review by brandonleung — COMMENTED** [review]
good job! added some nits

File: src/components/ErrorModal.tsx (hunk 1 above)
Comment (Ulydev) [reply src/components/ErrorModal.tsx:16]: Shoot you're right :)

File: src/components/setchecker/ParaSetChecker.tsx (hunk 4 above)
Comment (Ulydev) [reply src/components/setchecker/ParaSetChecker.tsx:157]: formatting with prettier (the configuration in .prettierrc)

**Review by brandonleung — APPROVED** [review]

**Pull request merged**